stellargraph/mapper/node_mappers.py

```diff
@@ -1128,3 +1128,275 @@ def flow_from_dataframe(self, node_targets, shuffle=False):
+        self.As = [A.todense()[None, :, :] for A in As]
+
+        # Reshape all inputs to have batch dimension of 1
+        self.target_indices = np.reshape(
+            self.target_indices, (1,) + self.target_indices.shape
+        )
+
+        self.features = np.reshape(self.features, (1,) + self.features.shape)
+        self.inputs = [self.features, self.target_indices] + self.As
+
+        # Reshape all inputs to have batch dimension of 1
+        self.target_indices = np.reshape(
+            self.target_indices, (1,) + self.target_indices.shape
+        )
+        self.features = np.reshape(self.features, (1,) + self.features.shape)
+
+        if targets is not None:
+            self.targets = np.asanyarray(targets)
+            self.targets = np.reshape(self.targets, (1,) + self.targets.shape)
+        else:
+            self.targets = None
+
+    def __len__(self):
+        return 1
+
+    def __getitem__(self, index):
+        return self.inputs, self.targets
+
+
+class RelationalFullBatchNodeGenerator:
+    """
+    A data generator for use with full-batch models on relational graphs e.g. RGCN.
+
+    The supplied graph G should be a StellarDiGraph object that is ready for
+    machine learning. Currently the model requires node features to be available for all
+    nodes in the graph.
+    Use the :meth:`flow` method supplying the nodes and (optionally) targets
+    to get an object that can be used as a Keras data generator.
+
+    This generator will supply the features array and the adjacency matrix to a
+    full-batch Keras graph ML model.  There is a choice to supply either a list of sparse
+    adjacency matrices (the default) or a list of dense adjacency matrices, with the `sparse`
+    argument.
+
+    For these algorithms the adjacency matrices require pre-processing and the default option is to
+    normalize each row of the adjacency matrix so that it sums to 1.
+    For customization a transformation (callable) can be passed that
+    operates on the node features and adjacency matrix.
+
+    Example::
+
+        G_generator = RelationalFullBatchNodeGenerator(G)
+        train_data_gen = G_generator.flow(node_ids, node_targets)
+
+        # Fetch the data from train_data_gen, and feed into a Keras model:
+        # Alternatively, use the generator itself with model.fit_generator:
+        model.fit_generator(train_gen, epochs=num_epochs, ...)
+
+    Args:
+        G (StellarGraphBase): a machine-learning StellarGraph-type graph
+        name (str): an optional name of the generator
+        transform (callable): an optional function to apply on features and adjacency matrix
+            the function takes (features, Aadj) as arguments.
+        sparse (bool): If True (default) a list of sparse adjacency matrices is used,
+            if False a list of dense adjacency matrices is used.
+
+    """
+
+    def __init__(self, G, name=None, sparse=True, transform=None):
+
+        if not isinstance(G, StellarGraphBase):
```

Comment:
In the docstring above, it says that G should be StellarDiGraph, but here the check if for the more general StellarGraph. Shall we update this test to be a bit more strict?

Reply:
The paper only talks about using directed graphs (I think relational graphs are inherently directed?) but I don't see any reason why this wouldn't work for undirected graphs. I'll update the docstring